prow/config/config.go

```diff
@@ -1915,10 +1915,18 @@ func validateTriggering(job Presubmit) error {
 	return nil
 }
 
-func validateReporting(r Reporter) error {
+func validateReporting(j JobBase, r Reporter) error {
 	if !r.SkipReport && r.Context == "" {
 		return errors.New("job is set to report but has no context configured")
 	}
+	if !r.SkipReport {
```

Comment:
What happens when !SkipReport && len(j.Labels) == 0?

Reply:
My understanding is that labels are used to control voting and grouping jobs together to aggregate job reporting. Setting the report label to the empty string causes a job to report immediately when it completes without aggregation (https://github.com/kubernetes/test-infra/pull/15596).  Omitting the report label altogether causes the default label of "Code-Review" to be used (should default to "Verified" IMO).
So for all possible states of the gerrit report label (populated, empty string, omitted) we still comment to Gerrit. I think skip_report then can control if we comment at all, but we should probably only allow skip_report when the gerrit report label is set to the empty label since we need to comment to vote on a label.

I think @amwat and @krzyzacy know the most about Gerrit reporting and labels. Is the above accurate? Do you have thoughts on how this config should work?

Reply:
Yes, that summary is accurate. also +1 to default verified (that is the intent of that label: https://gerrit-review.googlesource.com/Documentation/config-labels.html#label_Verified) the only problem is that verified "-1" is blocking by default and code-review is not, maybe after we have this skip-report feature Verified could be a sane default.

In terms of config, I guess i'm not sure what we want to achieve by skip-report, specifically whether we just want to 
1. skip reporting on a label but still show up in comments, 
2. or just not report at all to gerrit in any way.

If 2. then we can just return false here when skip-report is set (and exclude it from the selector)
https://github.com/kubernetes/test-infra/blob/6e7f9252d5fae7d74fd9ffabe9175078240000ca/prow/crier/reporters/gerrit/reporter.go#L104-L105

If 1. then forcing gerrit-report label to be empty makes sense.

Reply:
Thank you for the reviews. I have updated the PR to allow `skip_report` only if the Gerrit report label is set to empty or is omitted. I think the ignoring of `skip_report` was caused by the following: 

https://github.com/kubernetes/test-infra/blob/b5246a13a917f003450af87d15a94ff2dbc711f1/prow/gerrit/adapter/adapter.go#L307-L309



Crier will not run any reporters for a job which has `skip_report` set:
https://github.com/kubernetes/test-infra/blob/b5246a13a917f003450af87d15a94ff2dbc711f1/prow/crier/controller.go#L254-L258